server/api/api.go

```diff
@@ -2119,6 +2124,201 @@ func (a *API) handleUploadFile(w http.ResponseWriter, r *http.Request) {
+	//
+	// ---
+	// produces:
+	// - application/json
+	// parameters:
+	// - name: teamID
+	//   in: path
+	//   description: Team ID
+	//   required: true
+	//   type: string
+	// - name: time_range
+	//   in: query
+	//   description: duration of data to calculate insights for
+	//   required: true
+	//   type: string
+	// - name: page
+	//   in: query
+	//   description: page offset for top boards
+	//   required: true
+	//   type: string
+	// - name: per_page
+	//   in: query
+	//   description: limit for boards in a page.
+	//   required: true
+	//   type: string
+	// security:
+	// - BearerAuth: []
+	// responses:
+	//   '200':
+	//     description: success
+	//     schema:
+	//       type: array
+	//       items:
+	//         "$ref": "#/definitions/BoardInsight"
+	//   default:
+	//     description: internal error
+	//     schema:
+	//       "$ref": "#/definitions/ErrorResponse"
+
+	vars := mux.Vars(r)
+	teamID := vars["teamID"]
+	userID := getUserID(r)
+	query := r.URL.Query()
+	timeRange := query.Get("time_range")
+
+	if !a.permissions.HasPermissionToTeam(userID, teamID, model.PermissionViewTeam) {
+		a.errorResponse(w, r.URL.Path, http.StatusForbidden, "Access denied to team", PermissionError{"access denied to team"})
+		return
+	}
+
+	auditRec := a.makeAuditRecord(r, "getTeamBoardsInsights", audit.Fail)
+	defer a.audit.LogRecord(audit.LevelRead, auditRec)
+
+	page, err := strconv.Atoi(query.Get("page"))
+	if err != nil {
+		a.errorResponse(w, r.URL.Path, http.StatusBadRequest, "error converting page parameter to integer", err)
+		return
+	}
+	perPage, err := strconv.Atoi(query.Get("per_page"))
+	if err != nil {
+		a.errorResponse(w, r.URL.Path, http.StatusBadRequest, "error converting per_page parameter to integer", err)
+		return
+	}
+
+	userTimezone, aErr := a.app.GetUserTimezone(userID)
+	if aErr != nil {
+		a.errorResponse(w, r.URL.Path, http.StatusBadRequest, "Error getting time zone of user", aErr)
+		return
+	}
+	userLocation, _ := time.LoadLocation(userTimezone)
+	if userLocation == nil {
+		userLocation = time.Now().UTC().Location()
+	}
+	// get unix time for duration
+	startTime := mmModel.StartOfDayForTimeRange(timeRange, userLocation)
+	boardsInsights, err := a.app.GetTeamBoardsInsights(userID, teamID, &mmModel.InsightsOpts{
+		StartUnixMilli: mmModel.GetMillisForTime(*startTime),
+		Page:           page,
+		PerPage:        perPage,
+	})
+	if err != nil {
+		a.errorResponse(w, r.URL.Path, http.StatusInternalServerError, "time_range="+timeRange, err)
+		return
+	}
+
+	data, err := json.Marshal(boardsInsights)
+	if err != nil {
+		a.errorResponse(w, r.URL.Path, http.StatusInternalServerError, "", err)
+		return
+	}
+
+	jsonBytesResponse(w, http.StatusOK, data)
+
+	auditRec.AddMeta("teamBoardsInsightCount", len(boardsInsights.Items))
+	auditRec.Success()
+}
+
+func (a *API) handleUserBoardsInsights(w http.ResponseWriter, r *http.Request) {
+	// swagger:operation GET /teams/{teamID}/boards/insights getUserBoardsInsights
```

Comment:
This doc path is a different path than what is registered above.  It should be `/users/me/boards/insights`